Gems/PhysX/Code/Source/EditorHeightfieldColliderComponent.cpp

```diff
@@ -413,4 +431,28 @@ namespace PhysX
         return AZ::Aabb::CreateNull();
     }
 
+    void EditorHeightfieldColliderComponent::UpdateHeightfieldMaterialSelection(const Physics::MaterialSelection& updatedMaterialSelection)
+    {
+        AzPhysics::SimulatedBody* simulatedBody = m_sceneInterface->GetSimulatedBodyFromHandle(m_attachedSceneHandle, m_staticRigidBodyHandle);
+        if (!simulatedBody)
+        {
+            return;
+        }
+
+        AzPhysics::StaticRigidBody* rigidBody = azdynamic_cast<AzPhysics::StaticRigidBody*>(simulatedBody);
+
+        if (rigidBody->GetShapeCount() != 1)
+        {
+            AZ_Error("UpdateHeightfieldMaterialSelection",
+                rigidBody->GetShapeCount() == 1, "Heightfield collider should have only 1 shape. Count: %d", rigidBody->GetShapeCount());
+            return;
+        }
+
+        AZStd::shared_ptr<Physics::Shape> shape = rigidBody->GetShape(0);
+        PhysX::Shape* physxShape = azdynamic_cast<PhysX::Shape*>(shape.get());
+        physxShape->SetMaterialSelection(updatedMaterialSelection);
```

Comment:
`SetMaterialSelection` is not necessary, please remove use this instead here:

````
AZStd::vector<AZStd::shared_ptr<PhysX::Material>> materials;
Physics::PhysicsMaterialRequestBus::Broadcast(
                &Physics::PhysicsMaterialRequestBus::Events::GetMaterials,
                updatedMaterialSelection,
                materials);
physxShape->SetMaterials(materials);
````

Reply:
yeah good point, no need to add Material Selection to the class that doesn't know about it. thanks!